build.gradle

```diff
@@ -19,6 +20,12 @@ sourceSets.main.java.srcDirs = ['src']
 sourceSets.main.resources.srcDirs = ['res']
 sourceSets.test.java.srcDirs = ['test']
 
+checkstyle {
+    configFile = file("${project.rootDir}/conf/checkstyle.xml")
```

Comment:
since checkstyle analysis is not needed on upstream, it is only used during build, you may set it's location in project_root/data directory